pkg/xero/invoice.go

```diff
@@ -1,14 +1,91 @@
 package xero
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"io"
 )
 
+type InvoiceType int
+
+const (
+	ACCPAY InvoiceType = iota // A bill – commonly known as an Accounts Payable or supplier invoice
+	ACCREC                    // A sales invoice – commonly known as an Accounts Receivable or customer invoice
+)
+
+func (it InvoiceType) String() string {
+	return [...]string{"ACCPAY", "ACCREC"}[it]
```

Comment:
I'm sorry what? 😂

Reply:
Haha, maybe amend to this? [https://go.dev/play/p/g6st6kaO2sn](https://go.dev/play/p/g6st6kaO2sn)

The ... specifies the length of the array to be fixed and then we check for out-of-bounds:

```go
func (it InvoiceType) String() string {
	types := [...]string{"ACCPAY", "ACCREC"}
	if int(it) < 0 || int(it) >= len(types) {
		return "UNKNOWN"
	}
	return types[it]
}
```